Deprecate use of request.url

chosak opened the pull request:
This change removes the custom `request.url` attribute that gets added by our `sheerlike.middleware.GlobalRequestMiddleware`, instead using [the built-in Django `HttpRequest.build_absolute_uri` method](https://docs.djangoproject.com/en/2.1/ref/request-response/#django.http.HttpRequest.build_absolute_uri).

This change also removes the `request.url_rule` attribute which is unused anywhere in the codebase.

It also removes an unused `ask/_share_links.html` template and properly escapes one URL in the `ask/share-links-spanish.html` template.

To test using a local server and a production dump:

1. To test the standard social media molecule, visit a blog page, for example [this page](http://localhost:8000/about-us/blog/new-research-report-student-loan-repayment-and-broader-household-borrowing/) and confirm that the social media links work properly.
2. To test the Spanish Ask social media links, visit a page, for example [this one](http://localhost:8000/es/obtener-respuestas/que-efecto-tendria-sobre-mi-credito-comparar-diferentes-ofertas-de-prestamos-para-vehiculos-es-763/) and confirm that the social media links work properly. Note especially that the first link (Facebook) uses a properly encoded URL.
3. To test the Spanish Ask print link, vist the same page and verify that the "Imprimir" link works properly.
4. To test the base template links, visit any page, for example [this one](http://localhost:8000/), and verify that the `<link rel="canonical">` and `<meta property="og:url">` links look correct.
5. @higs4281 @willbarton would you mind verifying the changes to the regs3k code and unit test?

## Checklist

- [X] PR has an informative and human-readable title
- [X] Changes are limited to a single goal (no scope creep)
- [X] Code can be automatically merged (no conflicts)
- [X] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [X] Passes all existing automated tests
- [X] Any _change_ in functionality is tested
- [X] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [X] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

anselmbradford commented on cfgov/jinja2/v1/_includes/ask/share-links-spanish.html:
```diff
@@ -1,3 +1,3 @@
-<a href="https://api.addthis.com/oexchange/0.8/forward/email/offer?url={{ request.url|urlencode }}{% if utm_campaign %}%3Futm_campaign={{ utm_campaign }}{% endif %}&title={{  page.question if page.question else 'CFPB - Obtener Respuestas' }}&pubid=ra-4da354ee346886d2" class="share-email addthis_button_email"><img src="{{ static('nemo/_/img/icon-email-retina.png') }}" alt="Share via email" class=" ...
+<a href="https://www.facebook.com/sharer.php?u={{ request.build_absolute_uri()|urlencode }}{% if utm_campaign %}%3Futm_campaign={{ utm_campaign }}{% endif %}" class="share-facebook"><img src="{{ static('nemo/_/img/icon-facebook-retina.png') }}" alt="Share on Facebook" class="icon-20"></a>
+<a href="https://twitter.com/share?url={{ request.build_absolute_uri()|urlencode }}{% if utm_campaign %}%3Futm_campaign={{ utm_campaign }}{% endif %}&text={{ tweet_text if tweet_text else 'CFPB - Obtener Respuestas' }}&via=CFPB&lang=es" class="share-twitter"><img src="{{ static('nemo/_/img/icon-twitter-retina.png') }}" alt="Share on Twitter" class="icon-20"></a>
+<a href="https://api.addthis.com/oexchange/0.8/forward/email/offer?url={{ request.build_absolute_uri()|urlencode }}{% if utm_campaign %}%3Futm_campaign={{ utm_campaign }}{% endif %}&title={{  page.question if page.question else 'CFPB - Obtener Respuestas' }}&pubid=ra-4da354ee346886d2" class="share-email addthis_button_email"><img src="{{ static('nemo/_/img/icon-email-retina.png') }}" alt="Share v ...
```
TRIVIAL: Probably should place a space around the pipe, as is done in the social-media.html in this PR `request.build_absolute_uri() | urlencode`

chosak replied:
@anselmbradford thanks, will do.

willbarton commented:
That Travis failure though, does this mean that in places where we use `build_absolute_uri` we will have to pass a`SERVER_NAME` to manually-constructed `HttpRequest()`? 

Other thing the Travis failure makes me thing, we don't use the Django test client often enough in places where we need to pass requests. We probably should.

chosak commented:
@willbarton good point. Generally I try to use [`RequestFactory`](https://docs.djangoproject.com/en/2.0/topics/testing/advanced/#the-request-factory) to make requests for tests; these include the META options, GET/POST, etc. Unfortunately these don't include middleware - for that you have to use the full Django test client.

I've fixed the broken test (and rebased); the change is [here](https://github.com/cfpb/cfgov-refresh/pull/4307/commits/bd628a142d8106da8db383d1d94f97bdc7d45cc4#diff-b641b2fbe2a94dbfc11084d89faeb30aL547).
